File: src/main/java/at/fhhagenberg/sqe/api/ElevatorControlSystemServiceImpl.java
package at.fhhagenberg.sqe.api;

import at.fhhagenberg.sqe.entity.BuildingFloor;
import at.fhhagenberg.sqe.entity.Elevator;
import org.jetbrains.annotations.NotNull;
import sqelevator.IElevator;
import at.fhhagenberg.sqe.entity.ElevatorControlSystem;
import com.google.inject.Inject;

import java.rmi.RemoteException;
import java.util.List;

public class ElevatorControlSystemServiceImpl implements ElevatorControlSystemService {

    private final IElevator elevatorControl;
    private final ElevatorService elevatorService;
    private final BuildingFloorService buildingFloorService;

    @Inject
    public ElevatorControlSystemServiceImpl(
            IElevator elevatorControl,
            ElevatorService elevatorService,
            BuildingFloorService buildingFloorService
    ) {
        this.elevatorControl = elevatorControl;
        this.elevatorService = elevatorService;
        this.buildingFloorService = buildingFloorService;
    }

    @Override
    @NotNull
    public ElevatorControlSystem get() throws RemoteException {
        long clockTick = elevatorControl.getClockTick();
        int floorHeight = elevatorControl.getFloorHeight();
        List<BuildingFloor> floors = buildingFloorService.getAll();
        List<Elevator> elevators = elevatorService.getAll();

        return new ElevatorControlSystem(clockTick, floorHeight, floors, elevators);
    }
}

File: src/main/kotlin/at/fhhagenberg/sqe/api/ElevatorControlSystemServiceImpl.kt
package at.fhhagenberg.sqe.api

import at.fhhagenberg.sqe.entity.ElevatorControlSystem
import com.google.inject.Inject
import sqelevator.ConnectableIElevator
import java.rmi.RemoteException

class ElevatorControlSystemServiceImpl @Inject constructor(
        private val elevatorControl: ConnectableIElevator,
        private val elevatorService: ElevatorService,
        private val buildingFloorService: BuildingFloorService
) : ElevatorControlSystemService {
    @Throws(RemoteException::class)
    override fun get(): ElevatorControlSystem {
        val clockTick = elevatorControl.clockTick
        val floorHeight = elevatorControl.floorHeight
        val floors = buildingFloorService.getAll()
        val elevators = elevatorService.getAll()
        return ElevatorControlSystem(clockTick, floorHeight, floors, elevators)
    }
}